pkg/cmd/pr/checkout/checkout.go

```diff
@@ -119,10 +121,14 @@ func checkoutRun(opts *CheckoutOptions) error {
 		cmdQueue = append(cmdQueue, []string{"git", "fetch", headRemote.Name, refSpec})
 
 		// local branch already exists
-		if _, err := git.ShowRefs("refs/heads/" + newBranchName); err == nil {
+		if _, err := git.ShowRefs("refs/heads/" + newBranchName); err == nil && !opts.Force {
 			cmdQueue = append(cmdQueue, []string{"git", "checkout", newBranchName})
 			cmdQueue = append(cmdQueue, []string{"git", "merge", "--ff-only", fmt.Sprintf("refs/remotes/%s", remoteBranch)})
 		} else {
+			// If forced, then delete local branch
+			if err == nil && opts.Force {
+				cmdQueue = append(cmdQueue, []string{"git", "branch", "-D", newBranchName})
```

Comment:
This will work, but I don't think we should delete the branch entirely just to reset it to an updated commit. Git gives us enough tools to simply reset the tip of the branch.

Instead of `git merge --ff-only`, resetting a branch to match upstream would be done with `git reset --keep <ref>` or similar. Similarly, on the `git fetch <remote> <remote-ref>:<local-ref>` call this would have to add the `--force` flag.

Does this make sense? Do you think you'd be able to make the changes?

Reply:
Great, just updated it to not delete, and only reset the tip!